### PR TITLE
[003] view job capabilities

### PR DESCRIPTION
**Pull request by TaylorJCrawford**

### CODE REVIEW

**Review by dcorry586 — APPROVED** [review]

File: views/JobTitles.ejs [hunk 1]
```diff
@@ -66,13 +66,20 @@
...
       <% if (jobs.length> 0) { %>
         <% for (let job of jobs) { %>
-          <li><a href="#" class="list-group-item list-group-item-action">
-              <%= job.jobName %>
-            </a></li>
+            <li>
+              <a href="#" class="list-group-item list-group-item-action">
+                <%= job.jobName %>
+                <% if (job.jobCapabilityName != 'undefined') { %>
+                  - (<%= job.jobCapabilityName %> )
```
Comment (glenn-hamilton) [inline views/JobTitles.ejs]: You have a extra space after the Job Capability Name and the ending `)`

File: controllers/JobsController.ts [hunk 2]
```diff
@@ -1,21 +1,38 @@
...
+        let capabilities: JobCapabilities[];
         const title:string= 'Job Roles';
+        let correctIdx = [];
 
         try {
             jobs = await jobService.getJobs();
+
+            capabilities = await jobService.getJobCapabilities();
+
+            var mergedObjects = mergeObjectsByKey(capabilities, jobs, 'jobCapabilityId');
```
Comment (glenn-hamilton) [inline controllers/JobsController.ts]: `mergeObjectsByKey` and `mergedObjects ` scares me

File: views/pages/jobTitles.ejs [hunk 3]
```diff
@@ -26,6 +26,13 @@
               <a href="/jobs/<%= job.jobId %>">
                 <%= job.jobName %> - <%= job.bandName %>
               </a>
+              <a href="">
```
Comment (glenn-hamilton) [inline views/pages/jobTitles.ejs]: These links seem a bit broken

image

image

File: utils/common.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+
+export function mergeObjectsByKey(array1: any, array2: any, key:any) {
```
Comment (glenn-hamilton) [inline utils/common.ts]: Please tell me this is left over from something else?

Comment (ghost) [reply utils/common.ts]: Removed this as was not used. It was taylor's old code

File: views/pages/jobTitles.ejs (hunk 3 above)
Comment (ghost) [reply views/pages/jobTitles.ejs]: Fixed

**Review by glenn-hamilton — APPROVED** [review]

**Pull request merged**